addons/common/functions/fnc_addToInventory.sqf

```diff
@@ -57,7 +72,34 @@ switch (_type select 0) do {
                     (uniformContainer _unit) addWeaponCargoGlobal [_classname, 1];
                 };
                 default {
-                    _unit addWeaponGlobal _classname;
+                    if (_canFitWeaponSlot) then {
+                        _unit addWeaponGlobal _classname;
+                    } else {
+                        private _containerWithSpace = "";
+                        {
+                            _x params ["_parameters", "_container"];
+
+                            if (_parameters call CBA_fnc_canAddItem) exitWith {
+                                _containerWithSpace = _container;
+                            };
+                        } forEach [
+                            [[_unit, _classname, 1, false, false, true], "backpack"],
+                            [[_unit, _classname, 1, false, true, false], "vest"],
+                            [[_unit, _classname, 1, true, false, false], "uniform"]
+                        ];
+
+                        switch (_containerWithSpace) do {
+                            case "vest": {
+                                (vestContainer _unit) addWeaponCargoGlobal [_classname, 1];
+                            };
+                            case "backpack": {
+                                (backpackContainer _unit) addWeaponCargoGlobal [_classname, 1];
+                            };
+                            case "uniform": {
+                                (uniformContainer _unit) addWeaponCargoGlobal [_classname, 1];
+                            };
+                        };
```

Comment:
I think you can get rid of `_containerWithSpace` and just use `uniformContainer _unit`, `vestContainer _unit`, and `backpackContainer _unit` instead of the "uniform", "vest", and "backpack" strings. Doing that, L83 would become: `_container addWeaponCargoGlobal [_classname, 1];`.

Reply:
Would not have thought of that, thank you!